enterprise/server/util/oci/oci.go

```diff
@@ -332,25 +332,53 @@ func fetchImageFromCacheOrRemote(ctx context.Context, digestOrTagRef gcrname.Ref
 		log.CtxInfof(ctx, "Anonymous user request, skipping manifest cache for %s", digestOrTagRef)
 	}
 	if canUseCache {
-		desc, err := puller.Head(ctx, digestOrTagRef)
-		if err != nil {
-			if t, ok := err.(*transport.Error); ok && t.StatusCode == http.StatusUnauthorized {
-				return nil, status.PermissionDeniedErrorf("cannot access image manifest: %s", err)
+		var hash gcr.Hash
+		var desc *gcr.Descriptor
+		// If the ref doesn't contain a digest, then we need to make a HEAD
+		// request in order to resolve it to a digest, since manifest AC entries
+		// are keyed by digest.
+		if digestRef, ok := digestOrTagRef.(gcrname.Digest); ok {
```

Comment:
I can look at the code, but I _think_ it is possible for a `Reference` to contain a digest without being a `Digest` struct. Either way, can you pull this logic out into a function?

Reply:
It looks like `ParseReference` is the only way to get a `Reference` (?) and when using that func it looks like we can properly cast to `Digest` in all the cases where the ref contains a sha256 https://go.dev/play/p/4YxHafH33t_8

I extracted a `getDigest` func - looks a little cleaner now :+1:

Reply:
Oh nice -- thank you, `ParseReference`. That's good enough for me, then.